src/org/opensolaris/opengrok/search/context/QueryMatchers.java

```diff
@@ -89,9 +90,19 @@ private void getTerms(Query query) {
             getTerm((TermQuery) query);
         } else if (query instanceof PrefixQuery) {
             getPrefix((PrefixQuery) query);
+        } else if (query instanceof RegexpQuery) {
```

Comment:
Actually only current change in QueryMatchers.java is enough to see matched parts of documents on search results page - that is what I understand you describe as 'descriptions in the listing of found docs'.
However similar change in Summarizer seems safe, whatever it really does. I'll fix it too and try to find out it's real purpose :)
Thank you for helping me with the fix!